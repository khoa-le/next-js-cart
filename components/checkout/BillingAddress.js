import { useState, useEffect, useCallback } from "react";
import fetch from "isomorphic-unfetch";
import useDataApi from "../../lib/hooks/useDataApi";
import useSetState from "../../lib/hooks/useSetState";
import { createGlobalStyle } from "styled-components";

const BillingAddress = ({ formData, handleInputChange }) => {
  const [regions, setRegions] = useState([]);
  const [cities, setCities] = useState([]);

  useEffect(() => {
    const data = useDataApi("/api/directory/region", {}).then(data =>
      setRegions(data)
    );
  }, []);

  const fetchCityByRegion = useCallback(region_id => {
    useDataApi("/api/directory/city?region_id=" + region_id, {}).then(data =>
      setCities(data)
    );
  }, []);

  const handleRegionChange = event => {
    const region_id = event.target.value;
    fetchCityByRegion(region_id);
    handleInputChange("regionId", region_id);
  };

  return (
    <>
      <div className="mb-4">
        <h2 className="h5">Địa chỉ nhận hàng</h2>
      </div>

      <div className="row mb-9">
        <div className="col-md-6">
          <div className="js-form-message mb-6">
            <label className="h6 small d-block text-uppercase">
              Tên
              <span className="text-danger">*</span>
            </label>

            <div className="js-focus-state input-group u-form">
              <input
                className="form-control u-form__input"
                type="text"
                onChange={event =>
                  handleInputChange("firstName", event.target.value)
                }
                value={formData.firstName}
                required=""
                placeholder="Jack"
                aria-label="Jack"
                data-msg="Please enter your frist name."
                data-error-class="u-has-error"
                data-success-class="u-has-success"
              />
            </div>
          </div>
        </div>

        <div className="col-md-6">
          <div className="js-form-message mb-6">
            <label className="h6 small d-block text-uppercase">
              Họ
              <span className="text-danger">*</span>
            </label>

            <div className="js-focus-state input-group u-form">
              <input
                className="form-control u-form__input"
                type="text"
                onChange={event =>
                  handleInputChange("lastName", event.target.value)
                }
                value={formData.lastName}
                required=""
                placeholder="Wayley"
                aria-label="Wayley"
                data-msg="Please enter your last name."
                data-error-class="u-has-error"
                data-success-class="u-has-success"
              />
            </div>
          </div>
        </div>

        <div className="w-100"></div>

        <div className="col-md-6">
          <div className="js-form-message mb-6">
            <label className="h6 small d-block text-uppercase">
              Email address
              <span className="text-danger">*</span>
            </label>

            <div className="js-focus-state input-group u-form">
              <input
                className="form-control u-form__input"
                type="email"
                onChange={event =>
                  handleInputChange("email", event.target.value)
                }
                value={formData.email}
                required=""
                placeholder="jackwayley@gmail.com"
                aria-label="jackwayley@gmail.com"
                data-msg="Please enter a valid email address."
                data-error-class="u-has-error"
                data-success-class="u-has-success"
              />
            </div>
          </div>
        </div>

        <div className="col-md-6">
          <div className="js-form-message mb-6">
            <label className="h6 small d-block text-uppercase">
              Số điện thoại
              <span className="text-danger">*</span>
            </label>

            <div className="js-focus-state input-group u-form">
              <input
                className="form-control u-form__input"
                type="text"
                onChange={event =>
                  handleInputChange("phone", event.target.value)
                }
                value={formData.phone}
                placeholder="+1 (062) 109-9222"
                aria-label="+1 (062) 109-9222"
                data-msg="Please enter your last name."
                data-error-class="u-has-error"
                data-success-class="u-has-success"
              />
            </div>
          </div>
        </div>

        <div className="w-100"></div>

        <div className="col-md-6">
          <div className="js-form-message mb-6">
            <label className="h6 small d-block text-uppercase">
              Tỉnh/Thành Phố
              <span className="text-danger">*</span>
            </label>

            <div className="js-focus-state input-group u-form">
              <select onChange={handleRegionChange} className="custom-select">
                <option value="">--Chọn Tỉnh/Thành Phố--</option>
                {regions &&
                  regions.map(region => (
                    <option key={region.region_id} value={region.region_id}>
                      {region.default_name}
                    </option>
                  ))}
              </select>
            </div>
          </div>
        </div>

        <div className="col-md-6">
          <div className="mb-6">
            <label className="h6 small d-block text-uppercase">
              Quận/Huyện
              <span className="text-danger">*</span>
            </label>

            <select
              className="custom-select"
              onChange={event => handleInputChange("cityId", event.target.value)}
            >
              <option value="">--Chọn Quận/Huyện--</option>
              {cities &&
                cities.map(city => (
                  <option key={city.city_id} value={city.city_id}>
                    {city.default_name}
                  </option>
                ))}
            </select>
          </div>
        </div>

        <div className="w-100"></div>

        <div className="col-md-12">
          <div className="js-form-message mb-6">
            <label className="h6 small d-block text-uppercase">
              Địa chỉ
              <span className="text-danger">*</span>
            </label>

            <div className="js-focus-state input-group u-form">
              <input
                className="form-control u-form__input"
                type="text"
                onChange={event =>
                  handleInputChange("address", event.target.value)
                }
                value={formData.address}
                required=""
                placeholder="470 Lucy Forks"
                aria-label="470 Lucy Forks"
                data-msg="Please enter a valid address."
                data-error-class="u-has-error"
                data-success-class="u-has-success"
              />
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
export default BillingAddress;
