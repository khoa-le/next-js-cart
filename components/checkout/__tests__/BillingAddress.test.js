import React from "react";
import BillingAddress from "../BillingAddress";
import { render, cleanup, act } from "@testing-library/react";
import { renderHook } from "@testing-library/react-hooks";
jest.mock("../../../lib/hooks/useDataApi");

const initialFormData = {
  firstName: "Khoa",
  lastName: "Le ",
  email: "khoaln6@gmail.com",
  phone: "0978181187",
  regionId: "533",
  cityId: "",
  address: "86 Nguyen Huu Cau",
  payment_method: "COD",
  comments: "",
};
describe("Billing Form Address", () => {
  let useEffect = React.useEffect;
  beforeAll(() => {
    React.useEffect = React.useLayoutEffect;
  });
  afterAll(() => {
    React.useEffect = useEffect;
  });
  test("should render form", async () => {
    await act(async () => {
      const handleInputChange = jest.fn();

      //const useDataApi = jest.fn();
      //useDataApi.mockReturnValue([Promise.resolve()]);

      const { container, getByText } = render(
        <BillingAddress
          formData={initialFormData}
          handleInputChange={handleInputChange}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
