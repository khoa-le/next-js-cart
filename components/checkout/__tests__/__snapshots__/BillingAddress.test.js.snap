// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billing Form Address should render form 1`] = `
<div>
  <div
    class="mb-4"
  >
    <h2
      class="h5"
    >
      Địa chỉ nhận hàng
    </h2>
  </div>
  <div
    class="row mb-9"
  >
    <div
      class="col-md-6"
    >
      <div
        class="js-form-message mb-6"
      >
        <label
          class="h6 small d-block text-uppercase"
        >
          Tên
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <div
          class="js-focus-state input-group u-form"
        >
          <input
            aria-label="Jack"
            class="form-control u-form__input"
            data-error-class="u-has-error"
            data-msg="Please enter your frist name."
            data-success-class="u-has-success"
            placeholder="Jack"
            type="text"
            value="Khoa"
          />
        </div>
      </div>
    </div>
    <div
      class="col-md-6"
    >
      <div
        class="js-form-message mb-6"
      >
        <label
          class="h6 small d-block text-uppercase"
        >
          Họ
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <div
          class="js-focus-state input-group u-form"
        >
          <input
            aria-label="Wayley"
            class="form-control u-form__input"
            data-error-class="u-has-error"
            data-msg="Please enter your last name."
            data-success-class="u-has-success"
            placeholder="Wayley"
            type="text"
            value="Le "
          />
        </div>
      </div>
    </div>
    <div
      class="w-100"
    />
    <div
      class="col-md-6"
    >
      <div
        class="js-form-message mb-6"
      >
        <label
          class="h6 small d-block text-uppercase"
        >
          Email address
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <div
          class="js-focus-state input-group u-form"
        >
          <input
            aria-label="jackwayley@gmail.com"
            class="form-control u-form__input"
            data-error-class="u-has-error"
            data-msg="Please enter a valid email address."
            data-success-class="u-has-success"
            placeholder="jackwayley@gmail.com"
            type="email"
            value="khoaln6@gmail.com"
          />
        </div>
      </div>
    </div>
    <div
      class="col-md-6"
    >
      <div
        class="js-form-message mb-6"
      >
        <label
          class="h6 small d-block text-uppercase"
        >
          Số điện thoại
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <div
          class="js-focus-state input-group u-form"
        >
          <input
            aria-label="+1 (062) 109-9222"
            class="form-control u-form__input"
            data-error-class="u-has-error"
            data-msg="Please enter your last name."
            data-success-class="u-has-success"
            placeholder="+1 (062) 109-9222"
            type="text"
            value="0978181187"
          />
        </div>
      </div>
    </div>
    <div
      class="w-100"
    />
    <div
      class="col-md-6"
    >
      <div
        class="js-form-message mb-6"
      >
        <label
          class="h6 small d-block text-uppercase"
        >
          Tỉnh/Thành Phố
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <div
          class="js-focus-state input-group u-form"
        >
          <select
            class="custom-select"
          >
            <option
              value=""
            >
              --Chọn Tỉnh/Thành Phố--
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="col-md-6"
    >
      <div
        class="mb-6"
      >
        <label
          class="h6 small d-block text-uppercase"
        >
          Quận/Huyện
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <select
          class="custom-select"
        >
          <option
            value=""
          >
            --Chọn Quận/Huyện--
          </option>
        </select>
      </div>
    </div>
    <div
      class="w-100"
    />
    <div
      class="col-md-12"
    >
      <div
        class="js-form-message mb-6"
      >
        <label
          class="h6 small d-block text-uppercase"
        >
          Địa chỉ
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <div
          class="js-focus-state input-group u-form"
        >
          <input
            aria-label="470 Lucy Forks"
            class="form-control u-form__input"
            data-error-class="u-has-error"
            data-msg="Please enter a valid address."
            data-success-class="u-has-success"
            placeholder="470 Lucy Forks"
            type="text"
            value="86 Nguyen Huu Cau"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
